test/validate_test.go

```diff
@@ -0,0 +1,277 @@
+					"scratch_disk.#":                                      &terraform.ResourceAttrDiff{New: "1"},
+					"scratch_disk.0.interface":                            &terraform.ResourceAttrDiff{New: "SCSI"},
+					"service_account.#":                                   &terraform.ResourceAttrDiff{New: "1"},
+					"service_account.0.scopes.#":                          &terraform.ResourceAttrDiff{New: "3"},
+					"service_account.0.scopes.12345678":                   &terraform.ResourceAttrDiff{New: "https://www.googleapis.com/auth/devstorage.read_only", NewExtra: "storage-ro"},
+					"service_account.0.scopes.23456789":                   &terraform.ResourceAttrDiff{New: "https://www.googleapis.com/auth/userinfo.email", NewExtra: "userinfo-email"},
+					"service_account.0.scopes.34567890":                   &terraform.ResourceAttrDiff{New: "https://www.googleapis.com/auth/compute.readonly", NewExtra: "compute-ro"},
+					"tags.#":                                              &terraform.ResourceAttrDiff{New: "2"},
+					"tags.45678901":                                       &terraform.ResourceAttrDiff{New: "bar"},
+					"tags.56789012":                                       &terraform.ResourceAttrDiff{New: "foo"},
+					"zone":                                                &terraform.ResourceAttrDiff{New: "us-central1-a"},
+				},
+			},
+		},
+		{
+			kind: "google_storage_bucket",
+			name: "foo_storage_bucket",
+			diff: &terraform.InstanceDiff{
+				Attributes: map[string]*terraform.ResourceAttrDiff{
+					"location":                   &terraform.ResourceAttrDiff{New: "EU", NewExtra: "EU"},
+					"name":                       &terraform.ResourceAttrDiff{New: "image-store-bucket"},
+					"storage_class":              &terraform.ResourceAttrDiff{New: "STANDARD"},
+					"website.#":                  &terraform.ResourceAttrDiff{New: "1"},
+					"website.0.main_page_suffix": &terraform.ResourceAttrDiff{New: "index.html"},
+					"website.0.not_found_page":   &terraform.ResourceAttrDiff{New: "404.html"},
+				},
+			},
+		},
+		{
+			kind: "google_sql_database_instance",
+			name: "foo_sql_database_instance",
+			diff: &terraform.InstanceDiff{
+				Attributes: map[string]*terraform.ResourceAttrDiff{
+					"database_version":            &terraform.ResourceAttrDiff{New: "POSTGRES_9_6"},
+					"name":                        &terraform.ResourceAttrDiff{New: "master-instance"},
+					"region":                      &terraform.ResourceAttrDiff{New: "us-central-1"},
+					"settings.#":                  &terraform.ResourceAttrDiff{New: "1"},
+					"settings.0.disk_autoresize":  &terraform.ResourceAttrDiff{New: "true"},
+					"settings.0.pricing_plan":     &terraform.ResourceAttrDiff{New: "PER_USE"},
+					"settings.0.replication_type": &terraform.ResourceAttrDiff{New: "SYNCHRONOUS"},
+					"settings.0.tier":             &terraform.ResourceAttrDiff{New: "db-f1-micro"},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
```

Comment:
Is the expectation that whenever we support a new TF kind, we should add a corresponding test case here?

Reply:
Yes. Otherwise we should expect there are some code there is not covered by test.